puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -0,0 +1,201 @@
+pipeline {
+    agent { label 'fast' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Test Matrix') {
+            parallel {
+                stage('ruby-2.5-postgres') {
+                    agent { label 'fast' }
+                    environment {
+                        RUBY = '2.5'
+                        GEMSET = 'ruby-2.5-postgres'
+                    }
+                    stages {
+                        stage("setup-2.5-postgres") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: 'develop'
+                                configureRVM(env.RUBY, env.GEMSET)
+                                databaseFile(gemset(env.GEMSET))
+                                configureDatabase(env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("npm-2.5-postgres") {
+                            steps {
+                                withRVM(['npm install'], env.RUBY, env.GEMSET)
+                            }
+                        }
+                        stage("unit-tests-2.5-postgres") {
+                            steps {
+                                withRVM(['bundle exec rake jenkins:unit jenkins:integration pkg:generate_source TESTOPTS="-v" --trace'], env.RUBY, env.GEMSET)
```

Comment:
Would this be a good opportunity to split away the integration tests from the unit tests? right now we have to wait for the slowest test to finish, which is always the integration tests since they run a lot longer - if we split them and run them in parallel, we could get results faster.

Reply:
I could give that a test while I await further reviews.

Reply:
I made that change, and I can't say how much faster it made this pipeline but genreally speaking from what I can see: 

 * test_develop: avg of 1hr 10 minutes
 * foreman-develop-test: avg of 50 minutes

So I am seeing roughly 20 minute gain on this job.

Reply:
that seems inline with the timings of the current tests, usually mysql and sqlite (which don't run integration tests) take around 50min and pg takes around 1:10. We can also get rid of any steps related to npm installation and asset compilation in other tests so we might see even a bit of extra improvement there (and save bandwidth on downloading npm packages).